Assets/scripts/Inputs/InputComponent.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class InputComponent 
+{
+    BaseController controller;
+
+    public InputComponent(BaseController controller) {
+        this.controller = controller;
+        InputManager.Instance.RegisterInputComponent(this);
+    }
+    public void InputUpdate(float forward,float direction,bool shoot,bool boost)
+    {
+        controller.Move(forward,direction);
+        if (shoot)
+        {
+            controller.Shoot();
+        }
+        if (boost)
+        {
+            controller.StartBoost();
+        }
+        else
```

Comment:
what if the boost is controlled by time - so it should stop after X seconds. the current implementation will start boost in frame t and stop it in frame t+1?

Reply:
actually I was implementing it like running like tilli press shift it will run else simple walk